SYCL/ESIMD/api/functional/ctors/ctor_vector_fp_extra.cpp

```diff
@@ -0,0 +1,56 @@
+//==------- ctor_vector_fp_extra.cpp  - DPC++ ESIMD on-device test ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// XRUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// XRUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: false
+// XFAIL: *
+// TODO The simd can't be constructed with sycl::half data type. The issue was
```

Comment:
@kbobrovs added support for half types in simd type just recently. Please try the newer compiler.

Reply:
BTW, it may be Ok for this test not to check if HW supports those _extra_ fp types because it only constructs an object and does not have any math operations, but in general such test would require such check before running on device, i.e. sycl::aspect::fp16

Reply:
Should we skip a test if device does not support any underlying type?
Should we do something with reference data if, for example `nan` is not supported by current device, but it should be in the reference data?
I have the following concern:
- We should do a few runtime verifications during reference values generation
- We should do verifications for each optional features in each test

Reply:
Perhaps, it can be guarded by runtime macro, something like:
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Reduction/reduction_nd_ext_half.cpp#L20
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Reduction/reduction_nd_ext_double.cpp#L20
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Reduction/reduction_nd_ext_type.hpp#L22

I am not sure what to do with 'nan' not supported on some device. The test should check common functionality.
If there is some device specific detail, it could be moved to a separate test that would be run only subset of devices (i.e. additional/special test would have more restrictions).

Reply:
I've added verification in this commit: 32fd77fef94faa173c8688a9040514d2bfe56b4b

lets move additional discussions out of scope of this PR because of this affects all other tests on simd constructors.
I suggest create another one PR that provides logic for verification of all required optional device features.

Reply:
Ok, I agree, this is more general issue affecting many tests that are already merged, and it deserves a separate PR.
I approve now.
Please do not forget to handle types like half in one of nearest future PRs.